pkg/plugins/git/generic/main_test.go

```diff
@@ -1,11 +1,72 @@
 package generic
 
 import (
+	"fmt"
 	"os"
 	"path/filepath"
 	"testing"
+
+	"github.com/stretchr/testify/assert"
 )
 
+func TestIsSimilarBranch(t *testing.T) {
+
+	type data struct {
+		branchA        string
+		branchB        string
+		workingDir     string
+		expectedResult bool
+		expectedError  error
+	}
+
+	type dataSet []data
+
+	dSet := dataSet{
+		{
+			branchA:        "main",
+			branchB:        "issue-285",
+			workingDir:     "../../../..",
+			expectedResult: false,
+			expectedError:  nil,
+		},
+		{
+			branchA:        "main",
+			branchB:        "main",
+			workingDir:     "../../../..",
+			expectedResult: true,
+			expectedError:  nil,
+		},
+		{
+			branchA:        "main",
+			branchB:        "doNotExist",
+			workingDir:     "../../../..",
+			expectedResult: false,
+			expectedError:  fmt.Errorf("reference not found"),
+		},
+	}
+
+	for id, d := range dSet {
+		got, err := IsSimilarBranch(
+			d.branchA,
+			d.branchB,
+			d.workingDir)
+
+		if !assert.Equal(t, err, d.expectedError) {
+			t.Errorf("[%v] Expected error '%v' but got '%v'",
+				id,
+				d.expectedError,
+				err)
+		}
+
+		if !assert.Equal(t, got, d.expectedResult) {
+			t.Errorf("[%v] Expected result '%v' but got '%v'",
+				id,
+				d.expectedResult,
+				got)
+		}
+	}
+}
```

Comment:
```suggestion
	for id, d := range dSet {
		t.Run(fmt.Sprint(id), func(t *testing.T) {
			got, err := IsSimilarBranch(
				d.branchA,
				d.branchB,
				d.workingDir)

			if d.expectedError != nil {
				assert.Equal(t, err, d.expectedError)
				return
			}

			require.NoError(t, err)
			assert.Equal(t, got, d.expectedResult)
		})
	}
```

- Use a ["sub-test"](https://pkg.go.dev/testing#hdr-Subtests_and_Sub_benchmarks) with the instruction `t.Run` to identify the dataSet
- When using the `assert` methods, there is no need to write the error message by yourself: it's the goal of using assertion (it takes care of the message .. and more... for you).
- There is also the `require + return` pattern for the error checking:
  - Thanks to the `t.Run(){}` block, you can use the `return` instruction to exit the block (and iterate to the next subtest / test). It allows you not having to write conditional when the error is expected by the test.
  - The `require.` is a library from testify (like assert) that behaves like `assert` but fails the sub-test/test immediatly if it is false

Reply:
Thanks @dduportal for the suggestion and the explanation